### PR TITLE
Add edit functionality to App

### PR DESCRIPTION
**Pull request by F-Kirchhoff**
- Add edit functionality to edit topics, needs and proposals
- Add editForm to TopicView and ProposalView 
- Add respecitve PATCH routes to express routes

### CODE REVIEW

File: src/app/components/Forms/EditForm.tsx [hunk 1]
```diff
@@ -30,10 +29,8 @@ export default function EditForm({
 }: EditFormProps): JSX.Element {
   const [data, setData] = useState<ContentType | DEFAULT>(content)
 
-  function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
-    event.preventDefault()
-
-    data && onSubmit(data)
+  function handleSubmit() {
+    if (data) onSubmit(data)
```
Comment (marwinburesch) [inline src/app/components/Forms/EditForm.tsx:33]: So you are certain that the submit event doesn't interfere with this?

**Review by marwinburesch — COMMENTED** [review]
🚀

**Review by derjubu — APPROVED** [review]
LGTM :-)

File: src/server/routes/proposals.ts [hunk 2]
```diff
@@ -61,6 +61,24 @@ proposals.post('/', async (req: Request, res: Response) => {
   res.send(msg)
 })
 
+proposals.patch('/', async (req: Request, res: Response) => {
```
[the remark this answers is not in the document]
Comment (F-Kirchhoff) [reply src/server/routes/proposals.ts:64]: actually I do, its defined when the route is used in the api.ts :3

**Pull request merged**